docs/DeveloperGuide.md

```diff
@@ -4,11 +4,212 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
+## Architecture
+
+![](https://lh3.googleusercontent.com/FwYOJJpyhNrHNAUjWo1grnsWCDHdKMosDWaJaI_3kqtVbfD8108yk0MeJaybK3ac0MOAL3EVKYYCUjSiAbBrwAT8wRBkPTj2UDfw3AdhPT0fB8twOxfRDfh3BGAawOUOUlt-rxSACvVOhvqRnFamqWM)
+
+The **Architecture Diagram** given above explains the high-level design of Duck.
+
+Given below is a quick overview of main components and how they interact with each other.
+
+<br />
+
+**Main components of the architecture**
+
+```Duck``` is responsible for:
+
+**At app launch:** 
+
+-   Executes the ```Ui#greetingMessage()``` operation.
+
+-   Initializes a new ArrayList tasks 
+
+-   Populates the ArrayList by executing the ```Storage#tryLoad()``` operation.
+
+-   Begins scanning for user inputs, and passing these inputs to ```Parser``` class.
+
+-   At shut down: Shuts down the components and invokes the ```Ui#exitMessage()``` operation.
+
+```Commons``` represents a collection of classes used by multiple other components, such as ```java.time.LocalDateTime```
+
+The rest of the App consists of four components.
+
+-   ```UI```: The UI of the App.
+
+-   ```Parser```: The command executor.
+
+-   ```Storage```: Reads data from, and writes data to, the hard disk.
+
+-   ```TaskList```: Holds the data of Duck in memory.
+
+```TaskList``` stores items that belong to 4 subclasses that draw upon a shared superclass.
+
+```Task```: The skeleton superclass of all Tasks. 
+
+-   ```ToDo```: Basic form of ```Task```
+
+-   ```Deadline```: Expands upon ```Task``` by storing a ```by``` variable
+
+-   ```Event```: Expands upon ```Task``` by storing ```start``` and ```end``` variable
+
+-   ```SchoolClass```: Expands upon ```Task``` by storing ```start``` and ```end``` variable, along with a ```className``` variable
+
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+This section describes how Duck is implemented and how its commands are executed.
+
+### Purge feature
+
+**Sequence Diagram**
+
+![image](https://user-images.githubusercontent.com/1620654/227128864-cfc9cac7-bc68-4962-ac45-77e1ae067db0.png)
```

Comment:
![image](https://user-images.githubusercontent.com/89584177/228759652-88ac852c-55d2-45b3-b2bc-9188a2c4ccca.png)
Why is there a -> arrow after 1.5:task